src/common/crypto/platform/openssl/crypto.cpp

```diff
@@ -478,21 +478,48 @@ expected::ExpectedString ExtractPublicKey(const Args &args) {
 	if (ret != OPENSSL_SUCCESS) {
 		return expected::unexpected(MakeError(
 			SetupError,
-			"Failed to extract the public key from: (" + args.private_key_path
+			"Failed to extract the public key from bio: (" + args.private_key_path
 				+ "): OpenSSL BIO write failed: " + GetOpenSSLErrorMessage()));
 	}
 
-	int pending = BIO_ctrl_pending(bio_public_key.get());
+	EVP_PKEY *evp_public_key = PEM_read_bio_PUBKEY(bio_public_key.get(), NULL, NULL, NULL);
```

Comment:
I assume this does not need freeing, since you're grabbing it from `bio_public_key`, but can it be made const then? Makes the intention a bit clearer.

That is, unless `PEM_write_bio_PUBKEY` below requires it to be non-const.

Reply:
It does need freeing, I was too quick. Thx!